crud/cfg.lua

```diff
@@ -0,0 +1,68 @@
+---- Module for CRUD configuration.
+-- @module crud.cfg
+--
+
+local checks = require('checks')
+local errors = require('errors')
+
+local stash = require('crud.common.stash')
+local stats = require('crud.stats')
+
+local CfgError = errors.new_class('CfgError', {capture_stack = false})
+
+local cfg_module = {}
+
+local function set_defaults_if_empty(cfg)
+    if cfg.stats == nil then
+        cfg.stats = false
+    end
+
+    return cfg
+end
+
+local cfg = set_defaults_if_empty(stash.get(stash.name.cfg))
+
+--- Configure CRUD module.
+--
+-- @function __call
+--
+-- @tab self
+--
+-- @tab[opt] opts
+--
+-- @bool[opt] opts.stats
+--  Enable or disable statistics collect.
+--  Statistics are observed only on router instances.
+--
+-- @return Copy of configuration table.
+--
+local function __call(self, opts)
+    checks('table', { stats = '?boolean' })
+
+    opts = opts or {}
+
+    if opts.stats ~= nil then
+        if opts.stats == true then
+            stats.enable()
+        else
+            stats.disable()
+        end
+
+        rawset(cfg, 'stats', opts.stats)
+    end
+
+    return self
+end
+
+local function __newindex()
+    CfgError:assert(false, 'Use crud.cfg{} instead')
+end
+
+cfg_module.cfg = setmetatable({}, {
+    __index = cfg,
+    __newindex = __newindex,
+    __call = __call,
+    __serialize = function() return cfg end
+})
```

Comment:
Unlike `pairs(box.cfg)`, `pairs(crud.cfg)` will now work. Not really a bit deal, since here we have no backward compatibility guarantees.

However maybe it worth to add `crud.cfg:pairs()` and mention it in the documentation: just to don't let users write a wrong code with `pairs(crud.cfg)` or `fun.iter(crud.cfg)`. Or add `crud.cfg:unwrap()`, which returns a pure table. Or `crud.cfg:copy()`, which returns a copy. Don't know.

Maybe it is not needed for anyone. OTOH, using `fun.chain()` to merge several configs is quite useful. Say, when you have a default one, one collected from env and one provided in arguments. But whether we can need a current config is such chains? Don't know.

Let's ignore it for now. It is not critical.

I also added some more thoughts into https://github.com/tarantool/tarantool/issues/4521, if you're interested.

Reply:
Just in case: let's add a comment regarding `pairs(crud.cfg)` into the code. And I think it worth to leave a short note in README.

Reply:
I have referenced the comment in a new issue https://github.com/tarantool/crud/issues/265

Reply:
Added a comment